Content.Server/SS220/EmergencyShuttleControl/EmergencyShuttleLockdownSystem.cs

```diff
@@ -0,0 +1,213 @@
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Player;
+using Robust.Shared.Utility;
+using System.Linq;
+
+namespace Content.Server.SS220.EmergencyShuttleControl;
+/// <summary>
+///     System that manages the cancellation of emergency shuttle call.
+/// </summary>
+public sealed class EmergencyShuttleLockdownSystem : EntitySystem
+{
+    [Dependency] private readonly EntityManager _entityManager = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly StationSystem _station = default!;
+    [Dependency] private readonly NavMapSystem _navMap = default!;
+    [Dependency] private readonly ChatSystem _chat = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly EmergencyShuttleSystem _emergency = default!;
+    [Dependency] private readonly RoundEndSystem _roundEnd = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<EmergencyShuttleLockdownComponent, ComponentStartup>(OnComponentStartup);
+        SubscribeLocalEvent<EmergencyShuttleLockdownComponent, ComponentShutdown>(OnComponentShutdown);
+
+        SubscribeLocalEvent<CommunicationConsoleCallShuttleAttemptEvent>(OnShuttleCallAttempt);
+
+        SubscribeLocalEvent<EmergencyShuttleLockdownComponent, EmergencyShuttleLockdownToggleActionEvent>(OnEmergencyShuttleLockdownToggleAction);
+        SubscribeLocalEvent<EmergencyShuttleLockdownComponent, EmergencyShuttleLockdownActivateActionEvent>(OnEmergencyShuttleLockdownActivateActionEvent);
+        SubscribeLocalEvent<EmergencyShuttleLockdownComponent, EmergencyShuttleLockdownDeactivateActionEvent>(OnEmergencyShuttleLockdownDeactivateActionEvent);
+
+        SubscribeLocalEvent<EmergencyShuttleLockdownComponent, UseInHandEvent>(OnUseInHand);
+    }
+
+    #region Handlers
+    private void OnComponentStartup(Entity<EmergencyShuttleLockdownComponent> ent, ref ComponentStartup args)
+    {
+        if (ent.Comp.IsActivatedOnStartup)
+            Activate(ent);
+    }
+
+    private void OnComponentShutdown(Entity<EmergencyShuttleLockdownComponent> ent, ref ComponentShutdown args)
+    {
+        if (ent.Comp.IsActivated)
+            Deactivate(ent);
+    }
+
+    private void OnShuttleCallAttempt(ref CommunicationConsoleCallShuttleAttemptEvent ev)
+    {
+        var lockdowns = _entityManager.AllComponents<EmergencyShuttleLockdownComponent>();
+        var temp = lockdowns.Where(x => x.Component.IsActivated);
+        if (temp.Count() > 0)
+        {
+            ev.Cancelled = true;
+            ev.Reason = Loc.GetString(temp.First().Component.WarningMessage);
+            return;
+        }
+    }
+
+    private void OnEmergencyShuttleLockdownToggleAction(Entity<EmergencyShuttleLockdownComponent> ent, ref EmergencyShuttleLockdownToggleActionEvent args)
```

Comment:
```suggestion
    private void OnToggleAction(Entity<EmergencyShuttleLockdownComponent> ent, ref EmergencyShuttleLockdownToggleActionEvent args)
```

По названию системы и компонента итак понятен контекст